lib/torrent.js

```diff
@@ -1730,6 +1730,7 @@ class Torrent extends EventEmitter {
       )
 
       const reconnectTimeout = setTimeout(() => {
+        if (this.destroyed) return
         const newPeer = this._addPeer(peer.addr, this.client.utp ? 'utp' : 'tcp')
         if (newPeer) newPeer.retries = peer.retries + 1
       }, ms)
```

Comment:
Although this check will certainly fix the problem, as you have suggested I would change the `reconnectTimeout` reference scope to class level, in the same way that it is done with `_rechokeIntervalId`. So this code block could be something like this:
```
this._reconnectTimeoutId = setTimeout(() => {
  const newPeer = this._addPeer(peer.addr, this.client.utp ? 'utp' : 'tcp')
  if (newPeer) newPeer.retries = peer.retries + 1
}, ms)
if (this._reconnectTimeoutId.unref) this._reconnectTimeoutId.unref()
```
and we will also need to add in line 89
```
this._reconnectTimeoutId = null
```
and also add in line 661
```
clearTimeout(this._reconnectTimeoutId)
```

Reply:
Haha, sorry for being busy and blocking this. You should definitely have just run with the changes! xD